include/common_package/common_node.hpp

```diff
@@ -19,29 +20,61 @@ using namespace std::chrono_literals;
 class CommonNode : public rclcpp::Node
 {
 public:
-  /**
-   * @brief Constructor for creating a new CommonNode.
-   * @param id Unique name of the node.
-   */
-  CommonNode(std::string id) : Node(id) {
-    // Create a publisher for the "heartbeat" topic
-    publisher_ = this->create_publisher<interfaces::msg::Heartbeat>("heartbeat", 10);
-
-    // Create a timer that sends a heartbeat message every 500ms
-    timer_ = this->create_wall_timer(500ms, std::bind(&CommonNode::timer_callback, this));
-  }
+    /**
+     * @brief Constructor for creating a new CommonNode.
+     * @param id Unique name of the node.
+     */
+    CommonNode(std::string id) : Node(id)
+    {
+        // Create a publisher for the "heartbeat" topic
+        heartbeat_publisher = this->create_publisher<interfaces::msg::Heartbeat>("heartbeat", 1);
+
+        // Create a timer that sends a heartbeat message every 500ms
+        heartbeat_timer = this->create_wall_timer(std::chrono::milliseconds(heartbeat_period_ms), std::bind(&CommonNode::heartbeat_timer_callback, this));
+    }
+
+    /**
+     * @brief Get the active status of the node.
+     *
+     * @return true if the node is active, false otherwise.
+     */
+    bool get_active() const
+    {
+        return node_active;
+    }
 
 protected:
-  bool active = false;  ///< Indicating if node is currently active and sending commands to interface node
+    /**
+     * @brief Activates the node.
+     * 
+     * This function sets the `node_active` flag to true, indicating that the node is active.
+     */
+    void activate()
+    {
+        node_active = true;
+        RCLCPP_DEBUG(this->get_logger(), "CommonNode::activate: Activated node");
```

Comment:
Angeben des Funktionsnamen schadet nicht, kann man aber auch über das logging format wie hier: https://docs.ros.org/en/humble/Tutorials/Demos/Logging-and-logger-configuration.html#console-output-formatting beschrieben ausgeben lassen.
Ist also etwas redundant.

Reply:
Joa, hast du grundsätzlich schon recht. Mach ich aber immer so xD

Reply:
Alles klar